### PR TITLE
Added StepperMotor Module

### PR DESCRIPTION
**Pull request by ragv**
**Added a module to simulate a Stepper Motor. Uses PyQt4 for GUI.**

This essentially uses `setPixmap` to update a label in a pyqt window with the image of gear and shaft ( The F.V of a StepperMotor ) rotated by an angle which corresponds to the position of the stepper motor. The position is either updated by calling any of the methods `rotate()`, `move_to()` and `move_by()` or by altering the states of the connectors set to it ( which are stored in the `leads` list ) and calling `trigger()` which checks for the change and corresponds it with the `SEQ` ( a list of valid sequence ) and updates the `angle`.

The added stepper motor is Bipolar and Dual mode ( Full Step mode and Half Step mode ).

### CODE REVIEW

**Comment by debugger22** [comment]
:+1: Even the #100 was your.

Regarding this component, we'll have to figure out a better place to put this. Let's not mix digital and analog things.

**Comment by ghost** [comment]
:D 

Hey I am thinking about tools ? This will be digital only simulation ! It is operated by rotating of the bits like we do in Ring Counter ! See [this](http://www.freescale.com/files/microcontrollers/doc/app_note/AN2974.pdf) [ Skip to start of page 10 ] !! :)

I'll use that as a guide to build this !

**Comment by debugger22** [comment]
Nice. I didn't know about that.

**Comment by rajat974** [comment]
@debugger22 do not mix as in ?

**Comment by coder006** [comment]
@rvraghav93 can you elaborate a bit more about what kind of inputs will you be giving and what kind of functionality are you exactly expecting for ? I might be able to help :P

**Comment by debugger22** [comment]
@coder006 [this](http://www.freescale.com/files/microcontrollers/doc/app_note/AN2974.pdf) might be helpful.

**Comment by ghost** [comment]
@coder006 I'd love your help ! the code so far is [this](https://gist.github.com/rvraghav93/2a39e98e57aca15a1cf9). 

Explanation : The 1st module will calculate the angle based on the rotation of bits (`data`). The second one will display an image of Front View of the gear and shaft rotated by that angle. In effect, if the bits are rotated at a high speed this should translate to a rotating effect in the image . I am stuck with the qt part Sudhanshu is a bit busy at the moment.... meanwhile if you feel like, please do fork this branch and edit it ! I'll push it here !

**Comment by ghost** [comment]
@coder006 Initial version of the code has been pushed. Feel free to fork it and work with the same ! :)

**Comment by ghost** [comment]
@debugger22 This is done :D 

But 3 things:
1. I need some one who's good at image editing to modify the image `stepper_motor.jpg` to make the rotation look smooth.
2. The GUI is slightly a mess. It could be better if someone with pyqt experience could look into it ... 
3. Also pl see if it has any other issues :)

**Comment by debugger22** [comment]
You'll have to `PyQt4` install command in `.travis.yml`.

File: BinPy/tools/StepperMotor.py [hunk 1]
```diff
@@ -0,0 +1,550 @@
...
+    def __init__(self, bound_to, name, index):
+        super(_SMDisplay, self).__init__(None)
+
+        # Setting variables
+        self.bound_to = bound_to
+        self.name = name
+        self.index = index
+
+        # Configuring the window:
+        #screen = QtGui.QDesktopWidget().screenGeometry(self)
```
Comment (debugger22) [inline BinPy/tools/StepperMotor.py]: You can either remove this line or just put a space after `#`

**Comment by debugger22** [comment]
Thanks!

**Pull request merged**